Documentation enhance for Calibration algorithm (#1094)
intel/neural-compressor: docs/source/calibration.md

# Calibration Algorithms in Quantization

1. [Introduction](#introduction)

> 2. [Calibration Algorithm](#calibration-algorithm)

Reviewer: ```suggestion
2. [Calibration Algorithms](#calibration-algorithms)
```

> 3. [Supported Framework Matrix](#supported-framework-matrix)

Reviewer: ```suggestion
3. [Support Matrix](#support-matrix)
```

## Introduction

Quantization proves beneficial in terms of reducing the memory and computational requirements of the model. Uniform quantization transforms the input value $x ∈ [β, α]$ to lie within $[−2^{b−1}, 2^{b−1} − 1]$, where $[β, α]$ is the range of real values chosen for quantization and $b$ is the bit-width of the signed integer representation. Calibration is the process of determining the $α$ and $β$ for model weights and activations. Refer to this [link](https://github.com/intel/neural-compressor/blob/master/docs/source/quantization.md#quantization-fundamentals) for more quantization fundamentals

## Calibration Algorithm

> There are three main methods of Calibration:

Reviewer: How about changing to `Currently, INC supports three popular calibration algorithms:`? as there are others algo, such as MovingAverage?

- Entropy: This method minimizes the KL divergence to reduce the information loss between full-precision and quantized data [^2]. Its primary focus is on preserving essential information.

- Percentile: This method only considers a specific percentage of values for calculating the range, ignoring the remainder which may contain outliers [^3]. It enhances resolution by excluding extreme values but still retaining noteworthy data.

## Supported Framework Matrix

<table>
<thead>
  <tr>
    <th rowspan="2">Framework</th>
    <th colspan="2">Supported calibration algorithm</th>
  </tr>
  <tr>
    <th>weight</th>
    <th>activation</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Pytorch</td>
    <td>minmax</td>
    <td>minmax, kl</td>
  </tr>
  <tr>
    <td>Tensorflow</td>
    <td>minmax</td>
    <td>minmax, kl</td>
  </tr>
  <tr>
    <td>MXNet</td>
    <td>minmax</td>
    <td>minmax, kl</td>
  </tr>
  <tr>
    <td>OnnxRuntime</td>
    <td>minmax</td>
    <td>minmax, kl, percentile</td>
  </tr>
</tbody>
</table>

> `kl` is used to represent the Entropy calibration algorithm in Intel® Neural Compressor.

> Calibration algorithm is in the tuning space of Intel® Neural Compressor. After tuning, a quantization model with the most suitable calibration algorithm will be returned. Please refer to [tuning_strategies.md](https://github.com/intel/neural-compressor/blob/master/docs/source/tuning_strategies.md#tuning-space) for more details.

Reviewer: ```suggestion
The calibration algorithm is one of the tuning items utilized by Intel® Neural Compressor auto-tuning. The accuracy-aware tuning process will select an appropriate algorithm. Please refer to [tuning_strategies.md](https://github.com/intel/neural-compressor/blob/master/docs/source/tuning_strategies.md#tuning-space) for more details.

```

# Calibration Algorithm in Quantization

1. [Introduction](#introduction)
2. [Calibration Algorithm](#calibration-algorithm)
3. [Supported Framework Matrix](#supported-framework-matrix)

## Introduction

Quantization proves beneficial in terms of reducing the memory and computational requirements of the model. Uniform quantization transforms the input value $x ∈ [β, α]$ to lie within $[−2^{b−1}, 2^{b−1} − 1]$, where $[β, α]$ is the range of real values chosen for quantization and $b$ is the bit-width of the signed integer representation. Calibration is the process of determining the $α$ and $β$ for model weights and activations. Refer to this [link](https://github.com/intel/neural-compressor/blob/master/docs/source/quantization.md#quantization-fundamentals) for more quantization fundamentals

## Calibration Algorithm

There are three main methods of Calibration:

> - MinMax: This is the simplest method to get the maximum and minimum of input values as $α$ and $β$ [^1]. This approach maintains the full range but often at the cost of compromising precision.

Reviewer 2: in my opinion, minmax is also effective and not  'at the cost of compromising precision'
